x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceSecretsIndex.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.indices.SystemIndexDescriptor;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+import static org.elasticsearch.index.mapper.MapperService.SINGLE_MAPPING_NAME;
+import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
+
+public class InferenceSecretsIndex {
+
+    private InferenceSecretsIndex() {}
+
+    public static final String INDEX_NAME = ".infer-secrets";
+    public static final String INDEX_PATTERN = INDEX_NAME + "*";
+
+    // Increment this version number when the mappings change
+    private static final int INDEX_MAPPING_VERSION = 1;
+
+    public static Settings settings() {
+        return Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-1")
+            .build();
+    }
+
+    /**
+     * Reject any unknown fields being added by setting dynamic mappings to
+     * {@code strict} for the top level object. A document that contains unknown
+     * fields in the document root will be rejected at index time.
+     *
+     * The {@code secrets} object
+     * have dynamic mappings set to {@code false} which means all fields will
+     * be accepted without throwing an error but those fields are not indexed.
+     *
+     * The reason for mixing {@code strict} and {@code false} dynamic settings
+     * is that {@code secrets} is defined by
+     * the inference services and therefore are not known when creating the
+     * index. However, the top level settings are known in advance and can
+     * be strictly mapped.
+     *
+     * If the top level strict mapping changes then the no new documents should
+     * be indexed until the index mappings have been updated, this happens
+     * automatically once all nodes in the cluster are of a compatible version.
+     *
+     * @return The index mappings
+     */
+    public static XContentBuilder mappings() {
+        try {
+            return jsonBuilder().startObject()
+                .startObject(SINGLE_MAPPING_NAME)
+                .startObject("_meta")
+                .field("version", Version.CURRENT)
```

Comment:
We shouldn't use `Version.CURRENT` here. The `Version` class is going away due to releasing arbitrary commits from the main branch in serverless.

This field can be hardcoded as "8.11.0", as it's only needed to make pre-8.11.0 nodes aware that the index is newer than they understand.

Similar to https://github.com/elastic/elasticsearch/blob/cc572fd92d0076c74550133bc9cc9b0fc0552233/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/MlIndexAndAlias.java#L53-L63